bastion/src/envelope.rs

```diff
@@ -0,0 +1,131 @@
+use crate::broadcast::Sender;
+use crate::context::BastionContext;
+use crate::message::{BastionMessage, Message, Msg};
+use crate::path::BastionPath;
+use std::sync::Arc;
+
+#[derive(Debug)]
+pub(crate) struct Envelope {
+    pub(crate) msg: BastionMessage,
+    pub(crate) sign: Signature,
+}
+
+/// A tuple containing `Msg` and its `Signature`.
+pub type SignedMessage = (Msg, Signature);
```

Comment:
Reaching to a tuple at the surface API won't be idiomatic with `msg.1` we might want to yield a struct back and can get fields by name. Preferably these fields can have short names.